### ISSUE TITLE
Add `spawn_fn` to Remote.

### ISSUE THREAD

**Issue by carllerche** [tokio-rs/tokio-core#57]
This issue currently represents the task of unifying "spawning" functions across `Handle`, `Remote` and potentially other executors (like CPU pool).

Currently `Handle::spawn` takes a future where as `Remote::spawn` takes a fn.

Also, there are other questions to consider, such as #77
### Original issue

It looks like `spawn_fn` already exists on `Handle`.

https://tokio-rs.github.io/tokio-core/tokio_core/reactor/struct.Handle.html#method.spawn_fn

**Comment by carllerche** [comment]
Oh, looks like Remote::spawn already is a fn! sorry :(

**Comment by carllerche** [comment]
Maybe `Remote::spawn` should be renamed `Remote::spawn_fn` for consistency?

**Comment by carllerche** [comment]
Another thought, maybe `Remote::spawn` could return a `Future` of the return value (like `thread::spawn`). There could be an additional variant `remote::fire(....)` that doesn't return a future?

**Comment by alexcrichton** [comment]
Yeah I think we should rename for consistency, although we've avoided in the past having a return value of a future for two reasons:
1. The semantics elsewhere when you drop the receiver are that the computation is canceled, which we can do but may not be the best default?
2. This can always be easily emulated with `oneshot`.

**Comment by carllerche** [comment]
1. We should probably revisit this. I liked where it was left off where cancellation is the default, but there is a `detach` fn that prevents the cancellation.
2. I've needed to add the oneshot boilerplate every time (as far as I can tell looking back), which makes me think that it would be nice to include it by default (with another function that does't maybe, like `fire`).
